templates/document.html.slim

```diff
@@ -46,19 +47,8 @@ html lang=(attr :lang, 'en' unless attr? :nolang)
           TeX: {#{eqnums_opt}}
           });
       script src='#{cdn_base}/mathjax/2.4.0/MathJax.js?config=TeX-MML-AM_HTMLorMML'
-    - case document.attr 'source-highlighter'
-    - when 'coderay'
-      - if (attr 'coderay-css', 'class') == 'class'
-        - if @safe >= Asciidoctor::SafeMode::SECURE || (attr? :linkcss)
-          link rel='stylesheet' href=normalize_web_path('asciidoctor-coderay.css', (attr :stylesdir, ''))
-        - else
-          style=Asciidoctor::Stylesheets.instance.coderay_stylesheet_data
-    - when 'pygments'
-      - if (attr 'pygments-css', 'class') == 'class'
-        - if @safe >= Asciidoctor::SafeMode::SECURE || (attr? :linkcss)
-          link rel='stylesheet' href=normalize_web_path('asciidoctor-pygments.css', (attr :stylesdir, ''))
-        - else
-          style=Asciidoctor::Stylesheets.instance.pygments_stylesheet_data(attr 'pygments-style')
+    - if (syntax_hl = document.syntax_highlighter) && (syntax_hl.docinfo? :head)
+      =(syntax_hl.docinfo :head, node, linkcss: linkcss)
```

Comment:
Doesn't this remove coderay and pygment support?

Reply:
No both are working with Asciidoctor Ruby.
The code is using the new syntax highlighter API. The method `document.syntax_highlighter` will get the syntax highlighter configured: https://github.com/asciidoctor/asciidoctor/tree/master/lib/asciidoctor/syntax_highlighter


![coderay](https://user-images.githubusercontent.com/333276/69794076-eb2e4c00-11c9-11ea-8e34-1f06c4570469.png)
![pygment](https://user-images.githubusercontent.com/333276/69794077-eb2e4c00-11c9-11ea-8421-cf30c52f131c.png)

Reply:
But it's not compatible with 1.5.8 so it breaks backward compatibility with Asciidoctor < 2.0.0

Reply:
I guess we could check if the method `document.syntax_highlighter` exists to use the new API or not?

Reply:
Let's revert this change, we can do a follow-up pull request to support the new syntax highlighter API.